supabase/migrations/20240522022118_create_protected_posts_table.sql

```diff
@@ -0,0 +1,13 @@
+-- alter posts table
+ALTER TABLE posts
+ADD COLUMN have_protected_content BOOLEAN;
+-- create protected_posts table
+CREATE TABLE protected_posts (
+  id SERIAL PRIMARY KEY,
+  content JSON,
+  FOREIGN KEY (id) REFERENCES posts(id)
+);
+-- enable row level security
+ALTER TABLE protected_posts ENABLE ROW LEVEL SECURITY;
+-- create policy
+CREATE POLICY "Enable access protected post only to auhtenticated" ON protected_posts TO authenticated USING (true);
```

Comment:
With this, authenticated users can create, update and delete the data too.
So we just need to enable authenticated users ability to read data. I suggest to modify it like this:

```sql
CREATE POLICY "Enable access protected post only to auhtenticated" ON protected_posts FOR SELECT USING (auth.role() = 'authenticated');
```